### ISSUE TITLE
Problems with POST and SNI when mod_spdy is enabled

### ISSUE THREAD

**Issue by digi24** [eousphoros/mod-spdy#22]
I am using this module on Debian and Apache 2.4.10.  My SSL config uses name based virtual hosts and server name indication (SNI).  Browser is Chrome 38.0.2125.0. 

As long as I stick to GET requests, everything works as expected and many POST requests work too.  But some of the XMLHttpRequests send by POST from my browser are not answered.  From the browser console the requests seem to be sent to the correct hostname.

Looking at the Apache logs, I see that these requests expired several minutes later, with error 408.  They ended up being handled  by the default SSL host, indicating that SNI was not used.  

So in other words: The browser sends a POST request, Apache ignores SNI, answers from the wrong host and the browser does not accept this answer, as it originates from the wrong host and breaks some of the Javascript policies.  

I cannot reproduce this problem every time with mod_spdy, but it does not seem to occur, when I turn off mod_spdy with `SpdyEnabled off`.

**Comment by coroaes** [comment]
I have POST issues with Apache 2.4.7 and mod-spdy and SNI. Im running Ubuntu 14.04 LTS, I have compiled with the branch 2.4.7. The POST request when spdy is active doesnt log, the request doesnt arrive at the server.

**Comment by promethe42** [comment]
+1 I have the same issue with the apache-2.4.7 branch
